var/spack/repos/builtin/packages/py-carputils/package.py

```diff
@@ -28,6 +28,10 @@ class PyCarputils(PythonPackage):
 
     depends_on("git", type=("build", "run"))
 
+    # cusettings:11: DeprecationWarning:
+    # The distutils package is deprecated and slated for removal in Python 3.12.
```

Comment:
The warning creates a failure? Or there is a failure and Spack greps the warning instead of the real issue? Can you upload the build log when it fails?

Reply:
Thanks for the help in reviewing this. The installation succeeds, it is when using the tool in the dependent package opencarp at the installation phase that the issue is raised.

> ==> [2023-12-18-14:44:54.727545] 'cusettings' '/home/hk-project-test-fineci/scc-fineci-0001/.config/carputils/settings.yaml' '--flavor' 'ginkgo' '--software-root' '/hkfs/home/project/hk-project-test-fineci/scc-fineci-0001/spack/opt/spack/linux-rhel8-icelake/gcc-12.2.0/opencarp-13.0.ginkgo_integration-3r6qytm4p6tm4ysotfe3e62cfcicba5c/bin'
/hkfs/home/project/hk-project-test-fineci/scc-fineci-0001/spack/opt/spack/linux-rhel8-icelake/gcc-12.2.0/py-carputils-oc13.0-ayzyowiahlqtyufw2xky27m5dky2bj2w/bin/cusettings:11: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  import distutils.spawn
Generating new settings file /home/hk-project-test-fineci/scc-fineci-0001/.config/carputils/settings.yaml
Traceback (most recent call last):
  File "/hkfs/home/project/hk-project-test-fineci/scc-fineci-0001/spack/opt/spack/linux-rhel8-icelake/gcc-12.2.0/py-carputils-oc13.0-ayzyowiahlqtyufw2xky27m5dky2bj2w/bin/cusettings", line 195, in <module>
    new(args.filename, **kwargs)
  File "/hkfs/home/project/hk-project-test-fineci/scc-fineci-0001/spack/opt/spack/linux-rhel8-icelake/gcc-12.2.0/py-carputils-oc13.0-ayzyowiahlqtyufw2xky27m5dky2bj2w/lib/python3.11/site-packages/carputils/settings/config.py", line 372, in new
    write_settings_yaml(default_settings(*args, **kwargs), filename)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/hkfs/home/project/hk-project-test-fineci/scc-fineci-0001/spack/opt/spack/linux-rhel8-icelake/gcc-12.2.0/py-carputils-oc13.0-ayzyowiahlqtyufw2xky27m5dky2bj2w/lib/python3.11/site-packages/carputils/settings/config.py", line 295, in default_settings
    assert flavor in ['pt', 'petsc', 'direct'], "Choose flavor among: 'pt', 'petsc' or 'direct'"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Choose flavor among: 'pt', 'petsc' or 'direct'

Reply:
Can you update the comment to make this more clear? Do you know which file actually imports distutils? I don't see anything in git other than `bin/cusettings`, not sure if that gets installed.